site-src/guides/migrating-from-ingress.md

```diff
@@ -0,0 +1,431 @@
+# Migrating from Ingress
+
+The Gateway API project is the successor to the Ingress API. However, it does
+not include the Ingress resource (the closest parallel is the HTTPRoute). As a
+result, a one-time conversion from your existing Ingress resources to the
+relevant Gateway API resources is necessary.
+
+This guide will help you with the conversion. It will:
+
+* Explain why you may want to switch to the Gateway API.
+* Describe the key differences between the Ingress API and the Gateway API.
+* Map Ingress features to Gateway API features.
+* Show an example of an Ingress resource converted to Gateway API resources.
+* Mention [ingress2gateway](https://github.com/kubernetes-sigs/ingress2gateway)
+  for automatic conversion.
+
+At the same time, it will not prepare you for a live migration or explain how to
+convert some implementation-specific features of your Ingress controller.
+Additionally, since the Ingress API only covers HTTP/HTTPS traffic, this guide
+does not cover the Gateway API support for other protocols.
+
+## Reasons to Switch to Gateway API
+
+The [Ingress API](https://kubernetes.io/docs/concepts/services-networking/ingress/)
+is the standard Kubernetes way to configure external HTTP/HTTPS load balancing
+for Services. It is widely adopted by Kubernetes users and well-supported by
+vendors with many implementations ([Ingress controllers](https://kubernetes.io/docs/concepts/services-networking/ingress-controllers/))
+available. Additionally, several cloud-native projects integrate with the
+Ingress API, such as [cert-manager](https://cert-manager.io/)
+and [ExternalDNS](https://github.com/kubernetes-sigs/external-dns).
+
+However, the Ingress API has several limitations:
+
+- *Limited features*. The Ingress API only supports TLS termination and
+  simple content-based request routing of HTTP traffic.
+- *Reliance on annotations for extensibility*. The annotations approach to
+  extensibility leads to the limited portability and awkwardness of the API. 
```

Comment:
I feel like "awkwardness" here leaves a lot to decipher for the reader. Perhaps we can dig in more:

```suggestion
- *Reliance on annotations for extensibility*. The annotations approach to
  extensibility leads to limited portability as every implementation has its own
  supported extensions that may not translate to any other implementation.
```